### PR TITLE
Documents impersonation

### PR DESCRIPTION
**Pull request by klizhentas**

### CODE REVIEW

**Comment by klizhentas** [comment]
@r0mant @russjones please review when convenient

File: docs/pages/access-controls/guides/impersonation.mdx [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+Alice can login using `tsh` and issue a cert for `jenkins`:
+
+```bash
+$ tsh login --proxy=teleport.localhost:3080 --user=alice --auth=local
+$ tctl auth sign --user=jenkins --format=openssh --out jenkins --ttl=240h
+```
+
+Here is an example of how Alice can use the keys:
+
+```
```
Comment (r0mant) [inline docs/pages/access-controls/guides/impersonation.mdx:124]: bash

File: docs/pages/access-controls/guides/impersonation.mdx [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+
+- Even though Alice's `max_session_ttl` is 10 hours, she can issue a cert with a longer TTL of 240 hours,
+because `jenkins` role allows it. A certificate's TTL issued using impersonation is limited the `max_session_ttl` of the
+roles being impersonated.
+
+- Even in case if `jenkins` role could impersonate some other roles, Alice would not be able
+to use this permission. Teleport prevents impersonating users recursively.
+
+- Alice can get a new `jenkins` certificate with the same TTL and some fields updated.
+Teleport allows impersonated users to renew their certificates with with reduced the scope of the certificate.
```
Comment (r0mant) [inline docs/pages/access-controls/guides/impersonation.mdx:158]: with <strike>with</strike>

File: docs/pages/access-controls/guides/impersonation.mdx [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+To prevent unintended consequences, Teleport defines the following impersonation rules:
+
+- Even though Alice's `max_session_ttl` is 10 hours, she can issue a cert with a longer TTL of 240 hours,
+because `jenkins` role allows it. A certificate's TTL issued using impersonation is limited the `max_session_ttl` of the
+roles being impersonated.
+
+- Even in case if `jenkins` role could impersonate some other roles, Alice would not be able
+to use this permission. Teleport prevents impersonating users recursively.
+
+- Alice can get a new `jenkins` certificate with the same TTL and some fields updated.
```
Comment (r0mant) [inline docs/pages/access-controls/guides/impersonation.mdx:157]: It's not very clear what "some fields updated" means, I would give an example.

**Review by r0mant — APPROVED** [review]

File: docs/pages/access-controls/guides/impersonation.mdx [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+  name: alice
+spec:
+  traits:
+    group: ['security', 'devops']
+  roles:
+  - security-impersonator
+  - access
+```
+
+```bash
```
Comment (r0mant) [inline docs/pages/access-controls/guides/impersonation.mdx:275]: Also, this should probably show how to actually update the local `alice` user with those traits (the command).

**Pull request merged**